test/unit/modal.ts

```diff
@@ -0,0 +1,115 @@
+import { utils } from '@storefront/core';
+import Modal from '../../src/modal';
+import suite from './_suite';
+
+suite('Modal', ({ expect, spy, stub }) => {
+  let modal: Modal;
+
+  beforeEach(() => modal = new Modal());
+
+  describe('constructor()', () => {
+    describe('props', () => {
+      describe('autoOpen', () => {
+        it('should set default value', () => {
+          expect(modal.props).to.eql({ autoOpen: false });
+        });
+      });
+    });
+  });
+
+  describe('init()', () => {
+    it('should set visible state to autoOpen', () => {
+      const set = modal.set = spy();
+
+      modal.init();
+
+      expect(set).to.be.calledWithExactly({ visible: modal.props.autoOpen });
+    });
+  });
+
+  describe('handleOpen()', () => {
+    it('should set visible to true, add overflow: hidden style to body and add eventlistener', () => {
+      const preventDefault = spy();
+      const stopPropagation = spy();
+      const event: any = {
+        preventDefault,
+        stopPropagation,
+      };
+      const addEventListener = spy();
+      const style = { overflow: 'auto' };
+      const set = modal.set = spy();
+      stub(utils, 'WINDOW').returns({
+        document: {
+          addEventListener,
+          body: { style }
+        }
+      });
+
+      modal.handleOpen(event);
+
+      expect(preventDefault).to.be.calledOnce;
```

Comment:
I think `calledWithExactly()` might be better than `calledOnce`.

Reply:
not needed